src/object_accessor.hpp

```diff
@@ -90,41 +103,76 @@ void Object::set_property_value_impl(ContextType& ctx, const Property &property,
 
         ContextType child_ctx(ctx, property);
         List list(m_realm, *m_row.get_table(), col, m_row.get_index());
-        list.assign(child_ctx, value, try_update);
+        list.assign(child_ctx, value, try_update, update_only_diff);
         ctx.did_change();
         return;
     }
 
     switch (property.type & ~PropertyType::Nullable) {
         case PropertyType::Object: {
-            ContextType child_ctx(ctx, property);
-            auto link = child_ctx.template unbox<RowExpr>(value, true, try_update);
-            table.set_link(col, row, link.get_index(), is_default);
+            auto linked_schema = m_realm->schema().find(property.object_type);
```

Comment:
All of this code is just replicating what's already done by `unbox<RowExpr>()`. There shouldn't be any changes needed here other than forwarding on `update_only_diff` to `unbox()` and only calling `set_link` if it's changed.

Reply:
In the case where we link to objects with no primary key, we cannot just call `unbox<RowExpr>` with create flag set to true as this will unconditionally create a new object. I have rearranged the code to make it clearer.

Reply:
This has to call `unbox()` because there is binding-specific functionality which this is bypassing, with incorrect results. In Cocoa, the linked object may be in the form of an array dictionary of values, an unmanaged object, or a managed objects. For a dictionary of values, this does the correct thing. For an unmanaged object, this will fail to convert the object to a managed object. For a managed object which is not the object which the link currently points to, this will overwrite the values in the linked object rather than changing the link.

I think what this needs to do is if `update_only_diff` is `true`, it should pass the current value of the link to `unbox()`, and the `unbox()` would use that row rather than creating a new object when applicable.

Reply:
I am getting more and more insight into how the context concept is used :)
I have now created next attempt. A lot of parameters ...